app/common/expressions/managed.py

```diff
@@ -26,6 +27,20 @@ def statement(self) -> str:
     @abc.abstractmethod
     def description(self) -> str: ...
 
+    @property
+    @abc.abstractmethod
+    def message(self) -> str: ...
+
+    @property
+    def referenced_question(self) -> "Question":
+        # todo: split up the collections interface to let us sensibly reason about whats importing what
+        from app.common.data.interfaces.collections import get_question_by_id
+
+        # todo: this will do a database query per expression on the question - for now we'd anticipate
```

Comment:
As long as we're using db.session.get, SQLAlchemy's cache should short-circuit an actual DB query and just fetch the local object from cache. So this might look worse on the surface than it actually is.